centreon/src/Core/ResourceAccess/Application/UseCase/UpdateRule/UpdateRule.php

```diff
@@ -0,0 +1,483 @@
+    {
+        // Do not do uneccessary database calls if nothing has changed
+        if (
+            ! $this->shouldUpdateContactOrContactGroupRelations(
+                $rule->getLinkedContactIds(),
+                $updateRequest->contactIds
+            )
+        ) {
+            $this->validator->assertContactIdsAreValid($updateRequest->contactIds);
+
+            $this->debug(
+                'Deleting contacts - resource access rule relations',
+                ['id' => $updateRequest->id, 'contact_ids' => $rule->getLinkedContactIds()]
+            );
+            $this->writeRepository->deleteContactRuleRelations($updateRequest->id);
+
+            $this->debug(
+                'Creating contacts - resource access rule relations',
+                ['id' => $updateRequest->id, 'contact_ids' => $updateRequest->contactIds]
+            );
+            $this->writeRepository->linkContactsToRule($updateRequest->id, $updateRequest->contactIds);
+        }
+    }
+
+    /**
+     * @param Rule $rule
+     * @param UpdateRuleRequest $updateRequest
+     *
+     * @throws RuleException
+     * @throws AssertionFailedException
+     */
+    private function updateBasicInformation(Rule $rule, UpdateRuleRequest $updateRequest): void
+    {
+        // Do not do uneccessary database calls if nothing has changed
+        if ($this->shouldUpdateBasicInformation($rule, $updateRequest)) {
+            if ($rule->getName() !== NewRule::formatName($updateRequest->name)) {
+                $this->validator->assertIsValidName($updateRequest->name);
+                $rule->setName($updateRequest->name);
+            }
+
+            $rule->setIsEnabled($updateRequest->isEnabled);
+            $rule->setDescription($updateRequest->description);
+
+            $this->debug(
+                'Updating basic resource access rule information',
+                [
+                    'id' => $updateRequest->id,
+                    'name' => $rule->getName(),
+                    'description' => $rule->getDescription() ?? '',
+                    'is_enabled' => $rule->isEnabled(),
+                ]
+            );
+            $this->writeRepository->update($rule);
+        }
+    }
+
+    /**
+     * @param int[] $current
+     * @param int[] $update
+     *
+     * @return bool
+     */
+    private function shouldUpdateContactOrContactGroupRelations(array $current, array $update): bool
+    {
+        sort($current);
+        sort($update);
+
+        return array_diff($current, $update) === [];
+    }
+
+    /**
+     * @param Rule $current
+     * @param UpdateRuleRequest $updateRequest
+     *
+     * @return bool
+     */
+    private function shouldUpdateBasicInformation(Rule $current, UpdateRuleRequest $updateRequest): bool
+    {
+        return
+            $current->getName() !== NewRule::formatName($updateRequest->name)
+                || $current->getDescription() !== $updateRequest->description
```

Comment:
As the description is not a mandatory field what is the behaviour needed ?

If no description are filled in the request, should it be removed or should it be not modified ?

Reply:
With the comments above we can keep this as it is. If any of the 3 values contain changes then the basic information of the rule will need to be updated